stl/inc/sstream

```diff
@@ -100,6 +148,69 @@ public:
         _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
     }
 
+#if _HAS_CXX20
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    _NODISCARD basic_string<_Elem, _Traits, _Alloc2> str(const _Alloc2& _Al) const {
+        return basic_string<_Elem, _Traits, _Alloc2>{view(), _Al};
+    }
+
+    // The buffer cannot be moved to a string directly, because
+    // the buffer may already be full, and the terminating char is not '\0'.
+    // In that case, copy the string as usual.
+    _NODISCARD _Mystr str() && {
+        _Mystr _Result(_Al);
+        if (!(_Mystate & _Constant) && _Mysb::pptr()) { // writable
+            const auto _Base   = _Mysb::pbase();
+            const auto _Ptr    = _Mysb::pptr();
+            const auto _Eptr   = _Mysb::epptr();
+            const auto _Mysize = static_cast<_Mysize_type>((_STD max)(_Ptr, _Seekhigh) - _Base);
+            if (_Ptr == _Eptr
+                && !_Traits::eq(*(_Ptr - 1), _Elem())) { // the buffer is full and terminating char is not '\0'

```

Comment:
Should those `(_Ptr - 1)` be `_Prev_iter(_Ptr)`

It is hard to tell for me as I do not really know the type of `_Ptr`

Reply:
The type is `_Elem*` according to the standard. But you're right, it's better to use `_Prev_iter`. I'll change it.